### PR TITLE
[DateTime] Add style for week numbers

### LINKED ISSUE #1739: [Datetime] Missing week numbers style
### Bug report

- __Package version(s)__: 1.23.0

#### Steps to reproduce

Using the dayPickerProp showWeekNumbers the week numbers don't have the expected style:

bug

### PR DESCRIPTION
**Pull request by franciscofabian**
#### Checklist

- [x] [Enable CircleCI for your fork](https://circleci.com/add-projects)

### CODE REVIEW

**Comment by cmslewis** [comment]
Needs design input from @llorca. @franciscofabian can you add a screenshots of your changes for light theme and dark theme?

File: packages/datetime/src/_datepicker.scss [hunk 1]
```diff
@@ -109,6 +109,17 @@ $header-margin: ($header-height - $pt-input-height) / 2 !default;
...
 
+  .DayPicker-WeekNumber {
+    display: table-cell;
+    cursor: pointer;
+    min-width: 1rem;
+    padding: 0.5rem;
+    vertical-align: middle;
+    text-align: right;
+    color: $gray3;
+    font-size: 0.75em;
```
Comment (giladgray) [inline packages/datetime/src/_datepicker.scss]: we don't use `em` or `rem` elsewhere in the codebase so these values will likely have to be refactored.

**Comment by franciscofabian** [comment]
@cmslewis Sure, here are the changes in light and dark themes:

light
dark

File: packages/datetime/src/_datepicker.scss (hunk 1 above)
Comment (llorca) [reply packages/datetime/src/_datepicker.scss]:
```css.scss
width: $datepicker-day-size;
height: $datepicker-day-size;
font-size: $pt-font-size;

color: $pt-text-color-disabled;
// also for dark theme:
color: $pt-dark-text-color-disabled;
```

**Comment by franciscofabian** [comment]
<h3>fix</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/64/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/64/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/64/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

File: packages/datetime/src/_datepicker.scss [hunk 2]
```diff
@@ -112,12 +112,13 @@ $header-margin: ($header-height - $pt-input-height) / 2 !default;
   .DayPicker-WeekNumber {
     display: table-cell;
     cursor: pointer;
-    min-width: 1rem;
-    padding: 0.5rem;
+    width: $datepicker-day-size;
+    height: $datepicker-day-size;
+    padding-right: 10px;
```
Comment (cmslewis) [inline packages/datetime/src/_datepicker.scss]: General rule: use `$pt-grid-size` instead of `10px`.

Comment (franciscofabian) [reply packages/datetime/src/_datepicker.scss]: Changed, thanks!

**Comment by franciscofabian** [comment]
<h3>update</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/67/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/67/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/67/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

File: packages/datetime/src/_datepicker.scss [hunk 3]
```diff
@@ -109,6 +109,18 @@ $header-margin: ($header-height - $pt-input-height) / 2 !default;
     display: table-row;
   }
 
+  .DayPicker-WeekNumber {
+    display: table-cell;
+    cursor: pointer;
```
Comment (giladgray) [inline packages/datetime/src/_datepicker.scss]: hmm wondering about this line...

clicking on a week number produces a console error inside ReactDayPicker:

```
Uncaught TypeError: onWeekClick is not a function
    at onClick (Month.js:70)
    at Object.invokeGuardedCallback (ReactErrorUtils.js:26)
    at executeDispatch (EventPluginUtils.js:85)
    at Object.executeDispatchesInOrder (EventPluginUtils.js:108)
    at executeDispatchesAndRelease (EventPluginHub.js:43)
    at executeDispatchesAndReleaseTopLevel (EventPluginHub.js:54)
    at Array.forEach (<anonymous>)
    at forEachAccumulated (forEachAccumulated.js:24)
    at Object.processEventQueue (EventPluginHub.js:254)
    at runEventQueueInBatch (ReactEventEmitterMixin.js:17)
```

File: packages/datetime/src/_datepicker.scss [hunk 4]
```diff
@@ -109,6 +109,18 @@ $header-margin: ($header-height - $pt-input-height) / 2 !default;
...
+    width: $datepicker-day-size;
+    height: $datepicker-day-size;
```
Comment (giladgray) [inline packages/datetime/src/_datepicker.scss]: these two lines do not seem to have any effect. safe to remove?

File: packages/datetime/src/_datepicker.scss [hunk 5]
```diff
@@ -109,6 +109,18 @@ $header-margin: ($header-height - $pt-input-height) / 2 !default;
...
+    padding-right: $pt-grid-size;
+    vertical-align: middle;
+    text-align: right;
```
Comment (giladgray) [inline packages/datetime/src/_datepicker.scss]: hmm this right alignment combined with the right padding is an awkward effect. @llorca maybe we just want to remove padding and center text, like we do for actual dates?

File: packages/datetime/src/_datepicker.scss (hunk 3 above)
Comment (giladgray) [reply packages/datetime/src/_datepicker.scss]: something like that. might also be a bug in react-day-picker cuz it should handle optional callbacks? worth checking their release notes to see if we should update our dep.

Comment (llorca) [reply packages/datetime/src/_datepicker.scss]: `pointer-events: none` could do the trick too?

In any case, let's remove this and have a default cursor. No need for a pointer if it looks disabled and does nothing on click

File: packages/datetime/src/_datepicker.scss (hunk 5 above)
Comment (llorca) [reply packages/datetime/src/_datepicker.scss]: yes, just tested in the console: turns out we can remove `width`, `height`, `padding-right`, and change `text-align: right` to `text-align: center`

Comment (llorca) [reply packages/datetime/src/_datepicker.scss]: cc @franciscofabian

File: packages/datetime/src/_datepicker.scss (hunk 3 above)
Comment (franciscofabian) [reply packages/datetime/src/_datepicker.scss]: but onWeekClick can be added as a dayPickerProps so it can actually do something

Comment (llorca) [reply packages/datetime/src/_datepicker.scss]: right, but it doesn't do anything by default and looks disabled, which isn't good for interaction. is it possible to tweak the styling when `onWeekClick` is defined? not even sure that's a great idea, but we do need proper styling

**Comment by franciscofabian** [comment]
<h3>remove unnecessary styles</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/69/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/69/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/69/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

**Comment by franciscofabian** [comment]
<h3>prettier</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/75/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/75/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/franciscofabian/blueprint/75/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

File: packages/datetime/src/_datepicker.scss [hunk 6]
```diff
@@ -109,6 +109,15 @@ $header-margin: ($header-height - $pt-input-height) / 2 !default;
     display: table-row;
   }
 
+  .DayPicker-WeekNumber {
+    display: table-cell;
+    vertical-align: middle;
+    text-align: center;
+    color: $pt-text-color-disabled;
+    font-size: $pt-font-size;
+    pointer-events: none;
```
Comment (giladgray) [inline packages/datetime/src/_datepicker.scss]: I don't think we want this property... it breaks any support for `onWeekClick`.
But in the interest of velocity I think we can merge this PR and I can investigate the `onWeekClick` issues separately.

**Comment by franciscofabian** [comment]
Yes, sounds like a good idea for me.

**Comment by llorca** [comment]
sure if these are the same lines repeated 3 times

**Comment by giladgray** [comment]
yes I like that but we can do separately, along with https://github.com/palantir/blueprint/pull/1740#discussion_r146075603

**Comment by cmslewis** [comment]
@llorca @giladgray go ahead and approve if you're feeling good about this then. Would prefer two +1s. :)

**Comment by giladgray** [comment]
thanks @franciscofabian! I'll submit a PR momentarily with the follow-ups from comments.

**Comment by giladgray** [comment]
update: seems like `showWeekNumbers` and `onWeekClick` are meant to be set together and omitting the callback is actually undefined behavior. this means the whole `point-events` thing is moot and we should allow clicking on the element.

File: packages/datetime/src/_datepicker.scss (hunk 6 above)
Comment (cmslewis) [reply packages/datetime/src/_datepicker.scss]: @giladgray can you file an issue to track?

**Pull request merged**